### PR TITLE
Editorial: fix linking and change spelling of iterator interfaces

### LINKED ISSUE #3455: Uppercase "Iterator" falsely autolinks to constructor
#3395 introduced `#sec-iterator-constructor` as a link target:

Image

Should probably be changed to lowercase like here:

Image

### PR DESCRIPTION
**Pull request by michaelficarra**
Fixes #3455. This adds a bunch of `<dfn>`s and the auto-linking is sooo much better. Also makes a lot of progress getting rid of the remaining `<i>`s that are used as pseudo-`<dfn>`s.

**edit:** Actually, this eliminates *all* of the remaining `<i>`s! Nice!

### CODE REVIEW

File: spec.html [hunk 1]
```diff
@@ -3837,7 +3837,7 @@ <h1>Well-Known Intrinsic Objects</h1>
               <td>
               </td>
               <td>
-                The prototype of RegExp String Iterator objects (<emu-xref href="#sec-regexp-string-iterator-objects"></emu-xref>)
+                The prototype of RegExp String iterator objects (<emu-xref href="#sec-regexp-string-iterator-objects"></emu-xref>)
```
Comment (bakkot) [inline spec.html:3840]: I would prefer to leave `Iterator` capitalized when it's part of a name like this. "RegExp String Iterator" is a proper noun.

File: spec.html [hunk 2]
```diff
@@ -4041,7 +4041,7 @@ <h1>Well-Known Intrinsic Objects</h1>
               <td>
               </td>
               <td>
-                The prototype of wrapped Iterator objects returned by Iterator.from (<emu-xref href="#sec-%wrapforvaliditeratorprototype%-object"></emu-xref>)
+                The prototype of wrapped iterator objects returned by Iterator.from (<emu-xref href="#sec-%wrapforvaliditeratorprototype%-object"></emu-xref>)
```
Comment (bakkot) [inline spec.html:4044]: This use could remain capitalized as well, since the returned values do in fact inherit from `Iterator`.

Comment (michaelficarra) [reply spec.html:4044]: Elsewhere we use "Iterator instance objects" to refer to iterators in a way that also requires that they inherit from `Iterator`.

File: spec.html (hunk 1 above)
Comment (michaelficarra) [reply spec.html:3840]: I'm fine with that. Same with Array/String/Map/Set I'm assuming?

Comment (bakkot) [reply spec.html:3840]: Yeah.

File: spec.html [hunk 3]
```diff
@@ -42262,7 +42262,7 @@ <h1>Properties of Map Instances</h1>
 
     <emu-clause id="sec-map-iterator-objects">
       <h1>Map Iterator Objects</h1>
-      <p>A Map Iterator is an object, that represents a specific iteration over some specific Map instance object. There is not a named constructor for Map Iterator objects. Instead, map iterator objects are created by calling certain methods of Map instance objects.</p>
+      <p>A <dfn variants="Map iterators,Map iterator object,Map iterator objects">Map iterator</dfn> is an object, that represents a specific iteration over some specific Map instance object. There is not a named constructor for Map iterator objects. Instead, map iterator objects are created by calling certain methods of Map instance objects.</p>
```
Comment (bakkot) [inline spec.html:42265]: You did `s/object, that/object that/` on the other places this boilerplate appears; probably should do here as well.

File: spec.html (hunk 1 above)
Comment (michaelficarra) [reply spec.html]: Done.

**Comment by bakkot** [comment]
This also fixes https://github.com/tc39/ecma262/issues/3426

**Review by linusg — APPROVED** [review]
Nice, thanks!

**Review by jmdyck — CHANGES_REQUESTED** [review]
There are two remaining (non-`<h1>`) occurrences of "Iterator Objects" that you presumably want to change to "Iterator objects":
- `has properties that are inherited by all For-In Iterator Objects.`
- `has properties that are inherited by all Async-from-Sync Iterator Objects.`

**Review by bakkot — APPROVED** [review]

**Review by syg — APPROVED** [review]

**Pull request merged**